src/main/java/com/example/demo/controllers/ProductController.java

```diff
@@ -0,0 +1,61 @@
+package com.example.demo.controllers;
+
+import com.example.demo.dtos.ProductDTO;
+import com.example.demo.entities.User;
+import com.example.demo.mappers.DtoMapper;
+import com.example.demo.products.Product;
+import com.example.demo.services.UserService;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/products")
+public class ProductController {
+    private final UserService userService;
+    private final DtoMapper mapper;
+
+    @SneakyThrows
+    @GetMapping("/")
+    public List<? extends Product> currentUserProducts(@AuthenticationPrincipal User user, @RequestParam String productType){
+        Stream<Product> productStream = user.getProducts().stream().filter(product -> product.getClass().getName().equals(productType));
+        return userService.mapToType(productStream,productType);
+    }
+
+    @GetMapping("/{id}")
+    public List<? extends Product> otherUserProducts(@PathVariable Long id,@RequestParam String productType){
```

Comment:
the method above is secured, and this one isn't?

Reply:
It's a method for any user to look at other user's products (if they didn't hide them), that's why.